libraries/engage/core/helpers/i18n.js

```diff
@@ -0,0 +1,58 @@
+import {
+  getTranslator,
+  getPriceFormatter,
+  getDateFormatter,
+  getTimeFormatter,
+  getNumberFormatter,
+} from '@shopgate/pwa-common/helpers/i18n';
+import { logger } from '@shopgate/pwa-core/helpers';
+
+/**
+ * @typedef {Object} I18nHelpers
+ * @property {Function} init - Inits locales.
+ * @property {Function} text - Text translator.
+ * @property {Function} price - Price translator.
+ * @property {Function} date - Date translator.
+ * @property {Function} time - Time translator.
+ * @property {Function} number - Number translator
+ */
+/**
+ * I18n helpers.
+ * @returns {I18nHelpers}
+ */
+const I18n = () => {
+  let didInit = false;
+  /**
+   * Function that throws when called. Used in a default instance of i18n to notify developer
+   * when the module is used without initialization.
+   */
+  const notReadyCb = () => {
+    throw new Error('i18n is not yet ready');
+  };
+
+  return {
+    init({ locales, lang }) {
+      if (didInit) {
+        logger.warn('Looks like i18n locales are already inited. Changing locales now may lead to inconsistent translations.');
+      }
+      didInit = true;
+
+      this.text = getTranslator(locales, lang);
+      this.price = getPriceFormatter(lang);
+      this.date = getDateFormatter(lang);
+      this.time = getTimeFormatter(lang);
+      this.number = getNumberFormatter(lang);
+    },
+    text: notReadyCb,
```

Comment:
I would rather return original string if translation system is not ready yet.

Reply:
There is no translation string if system is not ready yet. Dictionary is empty in this case.

Reply:
Won't do as discussed earlier.